### PR TITLE
Reset enemy manager on game restart

### PR DESCRIPTION
**Pull request by SolomanDavis**
https://trello.com/c/sk7abhAQ

Added a method `EnemyManager::Restart()` that will reset the values in the enemy manager. This method should be called when the GameManager determines that it's time to restart the wave and therefore restart the enemy manager.

### CODE REVIEW

**Review by MV1090 — COMMENTED** [review]
This looks good. I just have a question, is there something in place that stops enemy's spawning if that enemy type is already full? for example if all the heavy enemy slots are full can it still spawn heavy enemy's? I might have missed it but wasn't sure. I'd also be happy to go through this at some point to better understand it all.

**Comment by SolomanDavis** [comment]
@MV1090 Not right now no, the way I have it set up, each `EnemySpawnInfo` contains the appropriate number of enemies to spawn. The idea here is that there will be the appropriate number of grid cells for each enemy type so that this isn't a problem.

If for some reason we have more enemies of a type than grid cells for that type to spawn, then I'd have to add some sort of logic as you say to ensure that no enemies of that type are spawned.

File: Assets/_Scripts/Managers/GameManager.cs [hunk 1]
```diff
@@ -85,15 +83,6 @@ protected override void Awake()
         EnemyManager.Instance.AllEnemiesKilled += OnAllEnemiesKilled;
     }
 
-    private void OnDisable()
```
Comment (EstelleSchoeman) [inline Assets/_Scripts/Managers/GameManager.cs:88]: Hi There,
Would you mind explaining why we are removing this?
I am struggling to understand.

Thank you

**Review by EstelleSchoeman — APPROVED** [review]
Good work.

File: Assets/_Scripts/Managers/GameManager.cs (hunk 1 above)
Comment (SolomanDavis) [reply Assets/_Scripts/Managers/GameManager.cs:88]: These were just comments I previously had that I let through into my previous pull request. I wanted to keep in mind that I'll need to figure out a way to clean up the EnemyManager appropriately when the scene ends but after today's discussions regarding how the EnemyManager is reset, this is actually no longer needed.

**Pull request merged**